administrator/components/com_associations/models/fields/itemlanguage.php

```diff
@@ -40,21 +40,24 @@ protected function getOptions()
 
 		$input = JFactory::getApplication()->input;
 
-		$referenceId         = $input->get('id', '');
-		$associatedComponent = $input->get('acomponent', '');
-		$associatedView      = $input->get('aview', '');
-		$extension           = $input->get('extension', '');
-		$realView            = $extension !== '' ? $extension : $associatedView;
+		$component           = $input->get('component', '', 'string');
+		$this->component     = AssociationsHelper::getComponentProperties($component);
+		$this->referenceId   = $input->get('id', 0, 'int');
+		$associatedComponent = $this->component->component;
```

Comment:
why save a var in another var? when yo ucould use just use $this->component->component below.

if ($associatedComponent != 'com_menus')

if ($this->component->component != 'com_menus')